bookshelter-app/src/layouts/index.tsx

```diff
@@ -2,16 +2,39 @@ import styles from './index.module.css';
 import Header from '../components/Header';
 import CategoriesSection from '../components/CategoriesSection';
 import CategoryList from '../components/CategoryList';
-
+import { useState, useEffect } from 'react';
+import { Categories } from '../types';
+import { get } from '../services/api';
+import { API } from '../constants/api';
 interface LayoutProps {
   children: React.ReactNode;
 }
+
 const DefaultLayout = ({ children }: LayoutProps) => {
+  const [categories, setCategories] = useState<Categories[]>([]);
+  const [currentCategory, setCurrentCategory] = useState<string | null>(null);
+  
+  useEffect(() => {
+    const fetchCategoriesList = async (): Promise<void> => {
+      const Categories = await get<Categories[]>(API.CATEGORIES_ENDPOINT);
+      if (Categories) setCategories(Categories);
+    };
+
+    fetchCategoriesList();
+  }, []);
+
+  const handleCategoryClick = (categoryName: string) => {
+    setCurrentCategory(categoryName);
+  };
+
   return (
     <>
       <Header />
-      <CategoriesSection />
-      <CategoryList />
+      <CategoriesSection currentCategory={currentCategory} />
+      <CategoryList
+        categories={categories}
+        onCategoryClick={handleCategoryClick}
```

Comment:
onClick is enough, similar update for other places

Reply:
Fixed!